src/Internal/DependencyManagement/ServiceProvider/PaymentsServiceProvider.php

```diff
@@ -84,13 +85,18 @@ public function register(): void {
 			->addArgument( HooksProxy::class )
 			->addArgument( LegacyProxy::class );
 
+		$container->addShared( PaymentFraudPreventionService::class )
```

Comment:
🤦 yeah, will do it. 

In my manual tests, there is no issue at all. Wonder that makes any difference? 🤔 Without providing this class in `$provides`, I can not statically it in `includes`, something like the following?

```php
 wcpay_get_container()->get( PaymentFraudPreventionService::class );
```

Reply:
So the logic is as follows:

1. Class is needed.
2. The container goes through all service providers, and checks if they provide the class.
3. If a service provider provides the class, its `register()` method is called once to register all the classes it provides.
4. Class is used.

> Wonder that makes any difference? 🤔

In most cases, we end up using another service from the same provider as the root. In this case, the `PaymentProcessingService` is always first, and it forces everything else to be loaded. In short: There is not much of a practical difference. It could still cause weird side-effects someday, so I guess it's better to have it.

Reply:
Resolved 96509c9acc58fa41706547206b5abe08a29efae1